### PR TITLE
Add C implementation for `@stdlib/math/base/special/exp`

### LINKED ISSUE #640: [RFC]: Add C implementation for `@stdlib/math/base/special/exp`
### Description

This RFC proposes adding a C implementation for [`@stdlib/math/base/special/exp`](https://github.com/stdlib-js/stdlib/tree/b43a75923e47775446d3b6d7219ee42e07f7ce63/lib/node_modules/%40stdlib/math/base/special/exp).

```c
double stdlib_base_exp( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #640 .

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] evalpoly.js
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

**Comment by Pranavchiku** [comment]
All tests and benchmarks cleared, this PR can have a review.

File: lib/node_modules/@stdlib/math/base/special/exp/README.md [hunk 1]
```diff
@@ -94,6 +94,94 @@ for ( i = 0; i < 100; i++ ) {
...
+```
+
+#### stdlib_base_exp( x )
+
+Evaluates the natural [exponential function][exponential-function].
+
+```c
+double out = stdlib_base_exp( 4.0 );
+// returns ~54.5982
+out = stdlib_base_exp( -9.0 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/exp/README.md:130]:
```suggestion

out = stdlib_base_exp( -9.0 );
```

File: lib/node_modules/@stdlib/math/base/special/exp/README.md [hunk 2]
```diff
@@ -94,6 +94,94 @@ for ( i = 0; i < 100; i++ ) {
...
+
+<section class="examples">
+
+### Examples
+
+```c
+#include "stdlib/math/base/special/exp.h"
+#include <stdlib.h>
+#include <stdio.h>
+int main() {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/exp/README.md:164]:
```suggestion

int main() {
```

File: lib/node_modules/@stdlib/math/base/special/exp/benchmark/c/native/benchmark.c [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+int main( void ) {
+	double elapsed;
+	int i;
+
+	// Use the current time to seed the random number generator:
+	srand( time( NULL ) );
+
+	print_version();
+	for ( i = 0; i < REPEATS; i++ ) {
+		printf( "# c::%s\n", NAME );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/exp/benchmark/c/native/benchmark.c:130]:
```suggestion
		printf( "# c::native::%s\n", NAME );
```

File: lib/node_modules/@stdlib/math/base/special/exp/src/main.c [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+* ```text
+* Copyright (C) 2004 by Sun Microsystems, Inc. All rights reserved.
+*
+* Developed at SunPro, a Sun Microsystems, Inc. business.
+* Permission to use, copy, modify, and distribute this
+* software is freely granted, provided that this notice
+* is preserved.
+* ```
+*/
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/exp/src/main.c:89]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/exp/src/main.c [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+/**
+* Computes \\(e^{r} 2^k\\) where \\(r = \mathrm{hi} - \mathrm{lo}\\) and \\(|r| \leq \ln(2)/2\\).
+*
+* @private
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/exp/src/main.c:93]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/exp/src/main.c [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+* @param hi - upper bound
+* @param lo - lower bound
+* @param k  - power of 2
+* @returns    function value
+*/
+double expmulti( const double hi, const double lo, const int32_t k ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/exp/src/main.c:99]:
```suggestion
static double expmulti( const double hi, const double lo, const int32_t k ) {
```

File: lib/node_modules/@stdlib/math/base/special/exp/src/main.c [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+*/
+
+/**
+* Computes \\(e^{r} 2^k\\) where \\(r = \mathrm{hi} - \mathrm{lo}\\) and \\(|r| \leq \ln(2)/2\\).
+*
+* @private
+* @param hi - upper bound
+* @param lo - lower bound
+* @param k  - power of 2
+* @returns    function value
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/exp/src/main.c:97]:
```suggestion
* @param hi     upper bound
* @param lo     lower bound
* @param k      power of 2
* @return       function value
```

File: lib/node_modules/@stdlib/math/base/special/exp/src/main.c [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+	r = hi - lo;
+	t = r * r;
+	c = r - (t * polyval_p( t ));
+	y = 1.0 - ( lo - ( ( r * c ) / ( 2.0 - c ) ) - hi);
+
+	return stdlib_base_ldexp( y, k );
+}
+
+// END: expmulti
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/exp/src/main.c:114]:
```suggestion
```

**Comment by kgryte** [comment]
As none of the proposed changes are blocking, I'll merge and touch up on my end.

**Pull request merged**